### PR TITLE
Project application functionnal

### LINKED ISSUE #37: Apply for Project
Since #32 is ready the application can adapt to allow a join request.

### PR DESCRIPTION
**Pull request by robsel118**
resolves #37 

- Users can follow and apply for projects 
- Project member can view applications
- Project member can accept invitation ( decline not implemented yet)
- Posting an existing skill will just return the corresponding skill
- created helper to check if member or owner
- user can sort project by tag or skills

### CODE REVIEW

**Comment by robsel118** [comment]
Will check it out

**Comment by MateyByrd** [comment]
> Will check it out

Found the bug, it was in the `authorization.js` file.

File: application/src/components/project-component/ApplicationList.vue [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      required: true
+    },
+    project: {
+      type: Object,
+      required: true
+    }
+  },
+  methods: {
+    redirectToUser (uid) {
+      window.location = `/profiles/${uid}`
```
Comment (MateyByrd) [inline application/src/components/project-component/ApplicationList.vue]:
```suggestion
      window.location = `/profile/${uid}`
```

**Comment by MateyByrd** [comment]
The functionality works! I think we can merge.

**Pull request merged**